keel-sql/src/main/resources/db/changelog/20200106-tie-artifacts-to-delivery-configs.yml

```diff
@@ -0,0 +1,91 @@
+databaseChangeLog:
+  - changeSet:
+      id: tie-artifacts-to-delivery-configs
+      author: emjburns
+      changes:
+        - addColumn:
+            tableName: delivery_artifact
+            columns:
+              - column:
+                  name: delivery_config_name
+                  type: varchar(255)
+                  value: "keel"
+                  constraints:
+                    nullable: false
+              - column:
+                  name: reference
+                  type: varchar(255)
+                  value: "keel"
+                  constraints:
+                    nullable: false
+      rollback:
+        - dropColumn:
+            tableName: delivery_artifact
+            columnName: delivery_config_name
+        - dropColumn:
+            tableName: delivery_artifact
+            columnName: reference
+  - changeSet:
+      id: create-new-artifacts-table
+      author: emjburns
+      changes:
+        - createTable:
+            tableName: artifact_versions
```

Comment:
Does this table not need a reference to the `delivery_artifact` table?

Reply:
Oh, I think I get it. `delivery_artifact` represents a subscription to an artifact by a particular `delivery_config` and this is a common list of versions. Is that right? If so do we need a reference from `delivery_artifact` to `delivery_config`?

Reply:
Ah, just saw @asher made the same comment above

Reply:
Nope, because we're storing versions only by name and type. Decoupling this allows us to keep track of versions independently of a delivery artifact. This means probably a little more storage of these versions, but is really a requirement when there are two artifacts that sort the versions in different ways. I can explain my thinking more offline if you'd like.

Reply:
Yes! @robfletcher you are right.